File: argentum-ui/src/app/common/rest-service/test-data/categories.ts
import {Category} from '../../model/category';

export namespace Categories {
  export const COAT_CHECK = new Category(11010, 'Coat check', '#ffff00')
  export const SOFT_DRINKS = new Category(11020, 'Soft drinks', '#00ffff');
  export const HARD_DRINKS = new Category(11030, 'Hard drinks', '#ff0000');

  export const ALL = [COAT_CHECK, SOFT_DRINKS, HARD_DRINKS];

  export const SPIRITS = new Category(11040, 'Spirits', '#ff00ff');
  export const SOFT_DRINKS_PATCHED_REQUEST = new Category(11021, 'Nonalcoholic', undefined);
  export const SOFT_DRINKS_PATCHED_RESPONSE = new Category(11021, 'Nonalcoholic', '#00ffff');
}

File: argentum-api/api/tests/data/categories.py
from api.models.category import Category
from api.tests.utils.test_objects import TestObjects


class TestCategories(TestObjects):
    MODEL = Category

    COAT_CHECK: MODEL
    SOFT_DRINKS: MODEL
    HARD_DRINKS: MODEL

    SPIRITS: MODEL
    SOFT_DRINKS_PATCHED: MODEL

    @classmethod
    def init(cls):
        cls.COAT_CHECK = cls.MODEL(
            id=11010,
            name='Coat check',
            color='#ffff00'
        )

        cls.SOFT_DRINKS = cls.MODEL(
            id=11020,
            name='Soft drinks',
            color='#00ffff',
        )

        cls.HARD_DRINKS = cls.MODEL(
            id=11030,
            name='Hard drinks',
            color='#ff0000',
        )

        cls.SAVED = [cls.COAT_CHECK, cls.SOFT_DRINKS, cls.HARD_DRINKS]

        cls.SPIRITS = cls.MODEL(
            id=11040,
            name="Spirits",
            color='#ff00ff',
        )

        cls.SOFT_DRINKS_PATCHED = cls.MODEL(
            id=11021,
            name='Nonalcoholic',
            color='#00ffff',
        )

        cls.UNSAVED = [cls.SPIRITS, cls.SOFT_DRINKS_PATCHED]
